@navikt/core/react/src/form/autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { Dispatch, SetStateAction, forwardRef, useState } from "react";
+import { createContext } from "../../util/create-context";
+import { useMergeRefs } from "../../util/hooks";
+import { createDescendantContext } from "../../util/hooks/descendants/useDescendant";
+import { SlottedDivElementRef } from "./SlottedDivElement";
+import { remove_virtual_focus, set_virtual_focus } from "./utils";
+
+// The recipe for success!
+//
+// 1. look at Menu PR, copy pasta + adjust
+// 2. useDescendants for switching focus (iterating focus through list)
+// 3. <Slot /> usage? it's (radix) the enabler for asChild (merge parent props into single child + melt away)
+// 4. <dismissableLayer /> click/focus outside
+// 5. <Floating /> utility
+//
+// perhaps a desired API?
+//
+// <Autocomplete>
+//    <Autocomplete.Anchor>
+//        ...
+//    </Autocomplete.Anchor>
+//    <Autocomplete.Content>
+//        <Autocomplete.Item>
+//           ...
+//        </Autocomplete.Item>
+//        <Autocomplete.Item>
+//           ...
+//        </Autocomplete.Item>
+//    <Autocomplete.Content>
+// </Autocomplete>
+
+export const [
+  AutocompleteDescendantsProvider,
+  useAutocompleteDescendantsContext,
+  useAutocompleteDescendants,
+  useAutocompleteDescendant,
+] = createDescendantContext<SlottedDivElementRef>();
+
+export const [AutocompleteContextProvider, useAutocompleteValue] =
+  createContext<{
+    value: string;
+    setValue: Dispatch<SetStateAction<string>>;
+  }>();
+
+const [AutocompleteInternalContextProvider, useAutocompleteInternalContext] =
+  createContext<{
+    virtualFocusIdx: number;
+    setVirtualFocusIdx: Dispatch<SetStateAction<number>>;
+  }>();
+
+export const Autocomplete = ({ children }: { children: React.ReactNode }) => {
+  const descendants = useAutocompleteDescendants();
+  const [value, setValue] = useState("");
+  const [virtualFocusIdx, setVirtualFocusIdx] = useState(0);
+  return (
+    <AutocompleteInternalContextProvider
+      {...{ virtualFocusIdx, setVirtualFocusIdx }}
+    >
+      <AutocompleteContextProvider {...{ value, setValue }}>
+        <AutocompleteDescendantsProvider value={descendants}>
+          <div
+            onChange={(event) => {
+              // assumption: there is a bubbling onChange from an input inside Anchor
+              setValue((event.target as HTMLInputElement).value);
+            }}
+          >
+            {children}
+          </div>
```

Comment:
Not 100% on this one, but should we assume that VirtualFocus will always be controlles? We dont need to have an internal state for `value` and `setState`, but rather assume that the user handles the input values.